tls/s2n_async_pkey.c

```diff
@@ -253,6 +253,9 @@ S2N_RESULT s2n_async_pkey_sign_async(struct s2n_connection *conn, s2n_signature_
     op->type = S2N_ASYNC_SIGN;
     op->conn = conn;
     op->validation_mode = conn->config->async_pkey_validation_mode;
+    if (conn->config->verify_after_sign) {
+        op->validation_mode = S2N_ASYNC_PKEY_VALIDATION_STRICT;
+    }
```

Comment:
Should this be added to the s2n_async_pkey_validation_mode documentation? Or is setting the validation mode to strict sort of obvious if you call s2n_verify_after_sign?

Reply:
I'd argue they're only related in the implementation :) s2n_async_pkey_validation_mode does exactly what it always did, and s2n_verify_after_sign does what its documentation says it does. They are just both paths to verifying signatures, so they happen to share an implementation now, but customers don't need to know that.

Reply:
Ah got it, okay.